lib/improver/utilities/cube_manipulation.py

```diff
@@ -685,3 +687,125 @@ def sort_coord_in_cube(cube, coord, order="ascending"):
     elif coord in ["height"] and order == "descending":
         cube.coord(coord).attributes["positive"] = "down"
     return cube[tuple(index)]
+
+
+def enforce_coordinate_ordering(
+        cube, coord_names, anchor="start", promote_scalar=False,
+        raise_exception=False):
+    """
+    Function to ensure that the requested coordinate within the cube are in
+    the desired position.
+
+    The reordering can either be anchored to the start or end of the available
+    dimension coordinates using the "anchor" keyword argument. If desired,
+    all the dimension coordinates can be reordered by specifying the coordinate
+    names in the desired order.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube where the ordering will be enforced to match the order within
+            the coord_names.
+        coord_names (list or str):
+            List of the names of the coordinates to order. If a string is
+            passed in, only the single specified coordinate is reordered.
+        anchor (str):
+            String to define where within the range of possible dimensions
+            the specified coordinates should be located. This could be either:
+            "start" or "end".
+        promote_scalar (bool):
+            If True, any coordinates that are matched and are not dimension
+            coordinates are promoted to dimension coordinates.
+            If False, any coordinates that are matched and are not dimension
+            coordinates will not be considered in the reordering.
+        raise_exception (bool):
+            Option as to whether an exception should be raised, if the
+            requested coordinate is not present.
+
+    Returns:
+        cube (iris.cube.Cube):
```

Comment:
The input cube is reordered. Did we decide if we need to mention when we are acting on the input cube directly?

Reply:
This has now been mentioned.